app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,103 @@
+class RecipesController < ApplicationController
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.all
+  end
+
+  def show
+    @recipe = Recipe.includes(:recipe_foods).find(params[:id])
+    @ingredients = @recipe.recipe_foods.where(recipe: @recipe)
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def edit; end
+
+  def create
+    @recipe = Recipe.new(recipe_params)
+    @recipe.user = current_user
+
+    respond_to do |format|
+      if @recipe.save
+        format.html { redirect_to recipe_url(@recipe), notice: 'Recipe was successfully created.' }
+        format.json { render :show, status: :created, location: @recipe }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /recipes/1 or /recipes/1.json
+  def update
+    respond_to do |format|
+      if @recipe.update(recipe_params)
+        format.html { redirect_to recipe_url(@recipe), notice: 'Recipe was successfully updated.' }
+        format.json { render :show, status: :ok, location: @recipe }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @recipe.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /recipes/1 or /recipes/1.json
+  def destroy
+    @recipe.destroy
+
+    respond_to do |format|
+      format.html { redirect_to recipes_url, notice: 'Recipe was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def new_ingredient
+    @recipe = Recipe.find(params[:recipe_id])
+    @food = Food.new
+    @recipe_food = @food.recipe_foods.build
+  end
+
+  def create_ingredient
+    @current_user = current_user
+    @recipe = Recipe.find(params[:recipe_id])
+    @food = @current_user.foods.build(food_params.except(:recipe_foods_attributes))
+
+    puts @food.inspect
+    if @food.save
+      @recipe_food = @food.recipe_foods.build(recipe: @recipe,
+                                              quantity: food_params[:recipe_foods_attributes]['0'][:quantity])
+
+      if @recipe_food.save
+        redirect_to recipe_path(@recipe), notice: 'Your Recipe is created successfully'
+      else
+        flash[:alert] = 'Something went wrong, try again!!'
+        puts @recipe_food.errors.full_messages
+        render :new_ingredient
+      end
+    else
+      flash[:alert] = 'Something went wrong, try again!!'
+      puts @food.errors.full_messages
+      render :new_ingredient
+    end
+  end
```

Comment:
- [ ] I couldn't create an ingredient(s) because the form data was rejected. Kindly fix this issue as this is the only way I can test generating the shopping list.

![ingredient](https://github.com/Odongo006/Recipe-App/assets/53124548/240ea608-e156-4a5b-8827-d37b384985b9)